traitsui/testing/tester/_dynamic_target_registry.py

```diff
@@ -0,0 +1,85 @@
+# (C) Copyright 2004-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Implement the interface of AbstractTargetRegistry to support
+dynamic checking on the target.
+"""
+import inspect
+
+from traitsui.testing.tester._abstract_target_registry import (
+    AbstractTargetRegistry,
+)
+from traitsui.testing.tester.exceptions import (
+    InteractionNotSupported,
+    LocationNotSupported,
+)
+
+
+class DynamicTargetRegistry(AbstractTargetRegistry):
+    """ A registry to support targets with dynamic checks.
+
+    Parameters
+    ----------
+    can_support : callable(target) -> bool
+        A callable that return true if the given target is supported
+        by the registry.
+    interaction_to_handler : dict(type, callable)
+        A dictionary mapping from interaction type to handler callables
+    """
+
+    def __init__(self, *, can_support, interaction_to_handler):
+        self.can_support = can_support
+        self.interaction_to_handler = interaction_to_handler
+
+    def _get_handler(self, target, interaction):
+        """ Reimplemented AbstractTargetRegistry._get_handler """
+        if interaction.__class__ not in self._get_interactions(target):
+            raise InteractionNotSupported(
+                target_class=target.__class__,
+                interaction_class=interaction.__class__,
+                supported=list(self._get_interactions(target)),
+            )
+        return self.interaction_to_handler[interaction.__class__]
```

Comment:
I understand the switch from `target_class` to `target`, but does `interaction_class` need to be `interaction` here (we only use `interaction.__class__`?  or is that just to be consistent?
e.g. in `_get_interaction_doc` we use the `interaction_class`

I don't think there is any problem with having it be `interaction`.  Just want to check I am not missing anything